client/allocrunner/alloc_runner_test.go

```diff
@@ -394,6 +394,89 @@ func TestAllocRunner_TaskMain_KillTG(t *testing.T) {
 	})
 }
 
+// TestAllocRunner_Lifecycle_Poststop asserts that a service job with 1
+// postop lifecycle hook starts all 3 tasks, only
+// the ephemeral one finishes, and the other 2 exit when the alloc is stopped.
+func TestAllocRunner_Lifecycle_Poststop(t *testing.T) {
```

Comment:
I would suggest changing the name of this test - as this tests stop behavior - so `TestAllocRunner_Lifecycle_Poststop_IfStopped`?

Also, should add a test if main tasks complete naturally, for batch and service jobs.
Also, should add a test for when